src/Plugins/Horoscope.php

```diff
@@ -57,23 +72,55 @@ public function divine(Command $command)
             );
         }
 
-        $dom = domdocument_load_html($response->getBody());
-        $xpath = new \DOMXPath($dom);
+        $globalHoroscopePageDom = domdocument_load_html($response->getBody());
+        $globalHoroscopeXPath = new \DOMXPath($globalHoroscopePageDom);
+
+        $currentHoroscopeUrl = $this->extractCurrentHoroscopeUrl($globalHoroscopeXPath);
+
+        /** @var HttpResponse $response */
+        $response = yield $this->httpClient->request($currentHoroscopeUrl);
+
+        if ($response->getStatus() !== 200) {
+            return $this->chatClient->postReply(
+                $command,
+                "The second stars are dead. It's better, but not quite there yet."
+            );
+        }
+
+        $currentHoroscopeDom = domdocument_load_html($response->getBody());
+        $currentHoroscopeXPath = new \DOMXPath($currentHoroscopeDom);
 
         if (!$command->hasParameters()) {
-            $sign = $this->extractActiveSign($xpath);
+            $sign = $this->extractActiveSign($currentHoroscopeXPath);
         }
 
         return $this->chatClient->postMessage(
             $command,
             $this->formatResponse(
                 $sign,
-                $this->extractDate($sign, $xpath),
-                $this->extractHoroscope($sign, $xpath)
+                $this->extractDate($sign, $currentHoroscopeXPath),
+                $this->extractHoroscope($sign, $currentHoroscopeXPath),
+                $currentHoroscopeUrl
             )
         );
     }
 
+    private function extractCurrentHoroscopeUrl(\DOMXPath $xpath): string
+    {
+        $url = $xpath->evaluate("
+            string(
+                //a[contains(@class,'js_entry-link')][1]
```

Comment:
When targeting classes it's generally better to normalize and exact match.  
Your current solution also matches classes like `foojs_entry-linkbar`.

What you want to do is something like `//a[contains(concat(" ", normalize-space(@class), " "), " js_entry-link ")][1]`

Reply:
Oh it looks like you refactored this piece out already.